compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -1249,6 +1249,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             return;
         }
 
+        // ignore suggestion to not make it broad in case of unallowed operations in let chains
+        // cc: https://github.com/rust-lang/rust/issues/147664
```

Comment:
```suggestion
        // Skip suggestion if LHS contains a let-chain at this would likely be spurious
        // cc: https://github.com/rust-lang/rust/issues/147664
```